### ISSUE TITLE
Py3 getxattr throws TypeError

### ISSUE THREAD

**Issue by joergwork** [fusepy/fusepy#34]
I think it's the same as #24 
fuse.py  line 558 

``` python
def getxattr(self, path, name, value, size, *args):
        ret = self.operations('getxattr', path.decode(self.encoding),
                                          name.decode(self.encoding), *args).encode(self.encoding)
```

seems to solve it.

**Comment by terencehonles** [comment]
Sorry for the super long delay. I had made a commit awhile ago referencing this issue, but I didn't update this issue.

Right now both `getxattr` and `setxattr` expect you to provide it `bytes` so the encoding process is up to you. The names are automatically encoded using the provided encoding which is why #24 was breaking. 

@eestrada we should discuss/research what should be encoded and what should be bytes. Part of my changes how the library functioned was due to the stricter nature of Python3. However this brought about the issue as discussed in #2
